static/app/utils/redirect.tsx

```diff
@@ -1,17 +1,27 @@
 import {useEffect} from 'react';
 import {InjectedRouter} from 'react-router';
-import {LocationDescriptor} from 'history';
+
+import {useNavigate} from './useNavigate';
 
 type Props = {
-  router: InjectedRouter;
-  to: LocationDescriptor;
+  to: string;
```

Comment:
This will have to be a string, instead of a LocationDescriptor, because useNavigate does not support `LocationDescriptorObject` which looks like this
```js
export interface LocationDescriptorObject {
    pathname?: Pathname;
    search?: Search;
    query?: QueryLike;
    hash?: Hash;
    state?: LocationState;
}
```

Reply:
@NisanthanNanthakumar is this change fine?